opennms-doc/guide-development/src/asciidoc/text/rest/measurements.adoc

```diff
@@ -160,3 +160,99 @@ curl -X POST  -H "Accept: application/json" -H "Content-Type: application/json"
     ]
 }
 ----
+===== More Advanced Expressions
+
+The JEXL 2.1.x library is used to parse the expression string and this also allows java objects and predefined functions to be included in the expression.
+
+JEXL uses a context which is pre-populated by OpenNMS with the results of the query.
+Several constants and arrays are also predefined as references in the context by OpenNMS.
+
+[%header,cols=2*] 
+|===
+|Constant or prefix
+|Description
+
+|__inf 
+|Double.POSITIVE_INFINITY
+
+|__neg_inf 
+|Double.NEGATIVE_INFINITY
+
+|NaN
+|Double.NaN
+
+|__E
+|java.lang.Math.E
+
+|__PI
+|java.lang.Math.PI
+
+|__diff_time
+|Time span between start and end of samples
+
+|__i
+|Index into the samples array which the present calculation is referencing
+
+|__AttributeName (where AttributeName is the searched for attribute)
+|This returns the complete double[] array of samples for AttributeName
+|===
+
+OpenNMS predefines a number of functions for use in expressions which are referenced by namespace:function
+
+Pre defined functions
+[%header,cols=3*] 
+|===
+|Function
+|Description
+|Example
+
+|jexl:evaluate("_formula"):
+|Passes a string to the JEXL engine to be evaluated as if it was entered as a normal expression. 
```

Comment:
Can we add a note here that the expected return value of the formula is a Double?